### PR TITLE
Proper TS types

### PR DESCRIPTION
**Pull request by Andarist**
Specifying unfetch type as the same as fetch is a stretch - so I've written down better, more accurate types.

### CODE REVIEW

**Comment by Andarist** [comment]
@developit friendly 🏓

**Comment by Andarist** [comment]
> This looks good to me, but I wonder if it'll cause folks trouble because the types are now different for unfetch VS isomorphic-unfetch?

Well, depends on how you look at it. It won't cause people much trouble because one is probably not using both in the same project, but I don't know how to properly support types from this PR when using `isomorphic-unfetch`. I'm not aware of any way to make TS resolve types based on target environment or anything like that - it always resolves node_modules to "main".

So landing this would definitely make both of those packages providing different typings, but at least for browser-only unfetch users they would be correct and would save users from using it like a regular, full-blown `fetch`.

**Comment by developit** [comment]
Makes sense. Anyone using `isomorphic-unfetch` is unaffected by this change then, they'll still get the (incorrect, but uniform) types from node-fetch?

**Comment by Andarist** [comment]
Yes - thats my understanding. Isomorphic-fetch wont even know about those typings here because they are not referenced anyhow from there

File: src/index.d.ts [hunk 1]
```diff
@@ -12,6 +12,33 @@ declare namespace unfetch {
   export type IsomorphicRequest = Request | NodeRequest
 }
 
-declare const unfetch: typeof fetch;
+type UnfetchResponse = {
```
Comment (1999) [inline src/index.d.ts]: Wouldn't it be better if we export this type so that TS developers could use it in their code?

File: src/index.d.ts [hunk 2]
```diff
@@ -12,6 +12,33 @@ declare namespace unfetch {
...
+	clone: () => UnfetchResponse,
+	headers: {
+		keys: () => string[],
+		entries: () => Array<[string, string]>,
+		get: (key: string) => string | undefined,
+		has: (key: string) => boolean,
+	}
+}
+
+type Unfetch = (
```
Comment (1999) [inline src/index.d.ts]: Same

**Comment by Andarist** [comment]
@developit friendly 🏓

**Review by developit — APPROVED** [review]

**Pull request merged**